src/app/app.component.ts

```diff
@@ -10,25 +16,95 @@ import { AuthService } from './auth/auth.service';
 export class AppComponent {
   title = 'altrusight';
   sidebarOpened: boolean;
+  allNonProfitsNames: string[] = [];
+
+  // searchInput = new FormControl();
+  // filteredNonProfits: Observable<string[]>;
+
 
   // TODO: issue with auth --> auth service is not instantiated until AFTER app component is
   // so we will always have auth issues on first render until fixing this
   // https://stackoverflow.com/questions/35191617/how-to-run-a-service-when-the-app-starts-in-angular-2
-  constructor(public router: Router, public authService: AuthService) {
+  constructor(public router: Router, public authService: AuthService, private nonProfitsService: NonprofitsService) {
       this.sidebarOpened = this.authService.isLoggedIn;
+      this.nonProfitsService.getNonprofit('')
+
+      // KEEP CODE BELOW: In case we want to improve search functinality
+      // as a drop down -->
+      // Initialize array that will hold nonprofit names
+      // this.initializeNonProfitsNamesArr();
+      // this.filteredNonProfits = this.searchInput.valueChanges.pipe(
+      //   // startWith(''),
+      //   map(value => this._filter(value))
+      // );
+  }
+
+  //========================================================
+  // Helper function to filer nonprofits as user types. 
+  // This function wil trigger on user input
+  //========================================================
+  private _filter(searchParam: string): string[] {
+    if(searchParam.length != 0) {
+
+      const filterValue = searchParam.toLowerCase();
+      return this.allNonProfitsNames.filter(option => option.toLowerCase().indexOf(filterValue) >= 0);    
+    }
+
+    return [];
+  }
+
+  //========================================================
+  // Will populate nonprofits names array with results of API
+  // call.
+  // TODO: API call returns only 100 nonprofits (in alphabetical order).
+  // Need to find a way to make this better. 
+  //========================================================
+  private initializeNonProfitsNamesArr(): void {
+
+    this.nonProfitsService.getNonprofits().subscribe((response) => {
+      for(var val of response) {
+        this.allNonProfitsNames.push(val.charityName);
+      }
+    });
   }
 
+  //========================================================
+  // Toggle side bar
+  //========================================================
   toggleSidebarOpened(): void {
     this.sidebarOpened = !this.sidebarOpened;
   }
 
+  //========================================================
+  // Will redirect web app to non-profits search page with 
+  // search parameter to display resutls
+  //========================================================
+  searchNonProfits(searchParam: string): void
+  {
+    // Only search if input is not empty
+    if(0 < searchParam.length )
```

Comment:
code would be cleaner with searchParam.length > 0 in condition

Reply:
[SUGGESTION] anoter thing about this logic though, once a user searches then they may want to return to the search page with "all" the nonprofits listed. perhaps have a flag for whether they previously searched and then want to search with an empty search to return all searchable nonprofits, so that users who want that behavior to work can have that, and users are still unable to search initially with an empty string

Reply:
If the user wants to return to the page with 'all' they can simply click 'nonprofits' --> 'all nonprofits' on the navbar.